### PR TITLE
[BACKLOG-5384, TEST, CLEANUP] - Increase unit test coverage

### PR DESCRIPTION
**Pull request by plagoa**
- Increased unit test coverage;
- Added integration tests;
- Applied CheckStyle to existing classes;

### CODE REVIEW

**Comment by plagoa** [comment]
@diogofscmariano @pamval please review.

File: src/org/pentaho/reporting/platform/plugin/connection/PentahoJndiConnectionReadHandler.java [hunk 1]
```diff
@@ -44,29 +44,26 @@ public PentahoJndiConnectionReadHandler() {
...
+   * @throws SAXException if there is a parsing error.
    */
   protected XmlReadHandler getHandlerForChild( final String uri, final String tagName, final Attributes atts )
-    throws SAXException {
+      throws SAXException {
     if ( isSameNamespace( uri ) == false ) {
       return null;
     }
-    if ( "path".equals( tagName ) ) { //$NON-NLS-1$
+    if ( "path" .equals( tagName ) ) { //$NON-NLS-1$
```
Comment (pamval) [inline src/org/pentaho/reporting/platform/plugin/connection/PentahoJndiConnectionReadHandler.java]: Is this correctly styled ?

Comment (plagoa) [inline src/org/pentaho/reporting/platform/plugin/connection/PentahoJndiConnectionReadHandler.java]: I applied the automatic rules for check style, but I'll reverse those specific changes.

**Comment by plagoa** [comment]
@pamval reviewed check style issue.

File: test-src/org/pentaho/reporting/platform/plugin/SimpleEmailComponentTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
+/*!
```
Comment (pamval) [inline test-src/org/pentaho/reporting/platform/plugin/SimpleEmailComponentTest.java]: Checkstyle issues in this file

File: test-src/org/pentaho/reporting/platform/plugin/SimpleEmailComponentTest.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  @Test
+  public void testSetOutputYpe() throws Exception {
+    sec.setOutputType( "outputType" );
+    assertEquals( "outputType", sec.getOutputType() );
+  }
+
+  @Test
+  public void testSetInputs() throws Exception {
+    final IContentItem contentItem = mock( IContentItem.class );
+    Map<String, Object> inputs = new HashMap<String, Object>() {{
```
Comment (pamval) [inline test-src/org/pentaho/reporting/platform/plugin/SimpleEmailComponentTest.java]: Here

File: test-src/org/pentaho/reporting/platform/plugin/SimpleReportingActionTest.java [hunk 4]
```diff
@@ -0,0 +1,341 @@
+/*!
```
Comment (pamval) [inline test-src/org/pentaho/reporting/platform/plugin/SimpleReportingActionTest.java]: Checkstyle issues in this file

File: test-src/org/pentaho/reporting/platform/plugin/StagingHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
+/*!
```
Comment (pamval) [inline test-src/org/pentaho/reporting/platform/plugin/StagingHandlerTest.java]: checkstyle issues

File: test-src/org/pentaho/reporting/platform/plugin/connection/PentahoMondrianConnectionProviderTest.java [hunk 6]
```diff
@@ -21,8 +21,12 @@
 
 import javax.sql.DataSource;
 import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.HashSet;
 import java.util.Properties;
+import java.util.Set;
```
Comment (pamval) [inline test-src/org/pentaho/reporting/platform/plugin/connection/PentahoMondrianConnectionProviderTest.java]: unused imports

File: test-src/org/pentaho/reporting/platform/plugin/drilldown/ContentLinkFunctionTest.java [hunk 7]
```diff
@@ -18,18 +18,123 @@
...
+import org.pentaho.reporting.libraries.formula.FormulaContext;
+import org.pentaho.reporting.libraries.formula.function.ParameterCallback;
 import org.pentaho.reporting.libraries.formula.function.userdefined.UserDefinedFunctionCategory;
+import org.pentaho.reporting.libraries.formula.lvalues.TypeValuePair;
+import org.pentaho.reporting.libraries.formula.typing.ArrayCallback;
+import org.pentaho.reporting.libraries.formula.typing.Sequence;
+import org.pentaho.reporting.libraries.formula.typing.TypeRegistry;
+import org.pentaho.reporting.libraries.formula.typing.Type;
 import org.pentaho.reporting.libraries.formula.typing.coretypes.AnyType;
 import org.pentaho.reporting.libraries.formula.typing.coretypes.TextType;
```
Comment (pamval) [inline test-src/org/pentaho/reporting/platform/plugin/drilldown/ContentLinkFunctionTest.java]: unused imports here

File: test-src/org/pentaho/reporting/platform/plugin/output/PageableHTMLOutputTest.java [hunk 8]
```diff
@@ -22,10 +22,16 @@
...
+import org.pentaho.reporting.libraries.base.config.ModifiableConfiguration;
+import org.pentaho.reporting.libraries.docbundle.DocumentBundle;
 import org.pentaho.reporting.libraries.resourceloader.ResourceManager;
 
 import java.io.OutputStream;
 
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+
 public class PageableHTMLOutputTest extends TestCase {
```
Comment (pamval) [inline test-src/org/pentaho/reporting/platform/plugin/output/PageableHTMLOutputTest.java]: more unused imports

**Comment by plagoa** [comment]
Fixed checkstyle issues, @pamval please review

File: test-src/org/pentaho/reporting/platform/plugin/output/ProxyOutputStreamTest.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+package org.pentaho.reporting.platform.plugin.output;
+
+import junit.framework.TestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.OutputStream;
+
+import static org.mockito.Mockito.*;
```
Comment (pamval) [inline test-src/org/pentaho/reporting/platform/plugin/output/ProxyOutputStreamTest.java]: Aren't star imports a checkstyle violation?

**Comment by plagoa** [comment]
yet again, @pamval please review.

**Pull request merged**